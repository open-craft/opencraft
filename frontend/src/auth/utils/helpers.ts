import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
