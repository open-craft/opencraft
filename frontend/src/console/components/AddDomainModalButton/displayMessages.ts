const messages = {
  buttonText: {
    defaultMessage: 'Add custom domain (€25/month)',
    description: 'Button for adding custom domain'
  },
  modalTitle: {
    defaultMessage: 'Add a Custom Domain',
    description: 'Title for custom domain modal'
  },
  modalDescriptionOne: {
    defaultMessage:
      "Add your domain name below. If you don't have a domain yet, we recommend you go to ",
    description: ''
  },
  gandiDomainText: {
    defaultMessage: 'Gandi.net',
    description: ''
  },
  modalDescriptionTwo: {
    defaultMessage:
      ' to order one. Then complete this step once you have a domain.',
    description: ''
  },
  externalDomain: {
    defaultMessage: 'Domain Name',
    description: 'Domain name input field'
  },
  addDomainBtn: {
    defaultMessage: 'Add Domain',
    description: 'Submit button to add domain'
  },
  domainRightsAgreement: {
    defaultMessage: 'I certify that I have the rights to use this domain',
    description: ''
  },
  additionalFeeAgreement: {
    defaultMessage: 'I agree to the additional monthly fee of €25',
    description: ''
  },
  domainErrorMessage: {
    defaultMessage: 'Invalid domain name',
    description: ''
  },
  domainErrorDescription1: {
    defaultMessage:
      "We couldnt recognize you domain name. If you're sure it's valid, please ",
    description: ''
  },
  contantSupport: {
    defaultMessage: 'contant support',
    description: ''
  },
  domainErrorDescription2: {
    defaultMessage: ' for assistance.',
    description: ''
  }
};

export default messages;
