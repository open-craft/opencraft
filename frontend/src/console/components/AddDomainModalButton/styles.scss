@import "~styles/theme";

.checkbox-label {
  font-size: 16px;
  color: $dark-2;
}

.add-domain-modal {
  padding: 35px;

  h2 {
    color: $dark-1;
  }

  .add-domain-modal-description {
    color: $dark-2;

    a {
      color: $link-blue;
    }
  }

  .domain-error {
    font-size: 16px;
    .error-msg {
      color: $alert-red-3;
      margin-bottom: 0.5rem;
    }
    a {
      color: $link-blue;
    }
  }

  button {
    font-size: 12px;
  }
}
