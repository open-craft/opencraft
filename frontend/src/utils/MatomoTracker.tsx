import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
