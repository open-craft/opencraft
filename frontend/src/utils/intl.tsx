import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
