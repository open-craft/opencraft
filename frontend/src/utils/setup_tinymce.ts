import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
