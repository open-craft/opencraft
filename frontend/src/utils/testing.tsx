import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
