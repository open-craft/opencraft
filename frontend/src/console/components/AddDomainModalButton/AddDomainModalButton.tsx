import { InstancesModel } from 'console/models';
import * as React from 'react';
import { Button, Col, Container, Form, Modal, Row } from 'react-bootstrap';
import { RootState } from 'global/state';
import { connect } from 'react-redux';
import { TextInputField } from 'ui/components';
import { WrappedMessage } from 'utils/intl';
import { updateFieldValue, clearErrorMessage } from 'console/actions';
import { SUPPORT_LINK } from 'global/constants';
import messages from './displayMessages';
import './styles.scss';

interface State {
  showModal: boolean;
  externalDomain?: string;
  domainRightsAggrement: boolean;
  additionalFeeAggrement: boolean;
}

interface ActionProps {
  updateFieldValue: Function;
  clearErrorMessage: Function;
}

interface StateProps extends InstancesModel {}

interface Props extends StateProps, ActionProps {}

class AddDomainButtonComponent extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      showModal: false,
      externalDomain: '',
      domainRightsAggrement: false,
      additionalFeeAggrement: false
    };

    if (props.activeInstance.data) {
      this.state = {
        showModal: false,
        externalDomain: props.activeInstance.data.externalDomain,
        domainRightsAggrement: false,
        additionalFeeAggrement: false
      };
    }
  }

  public componentDidUpdate(prevProps: Props) {
    const instance = prevProps.activeInstance;
    if (!instance.data) {
      this.handleHide();
    } else if (
      prevProps.activeInstance!.data!.externalDomain !==
      this.props.activeInstance!.data!.externalDomain
    ) {
      this.handleHide();
    }
  }

  private handleShow = () => {
    this.setState({
      showModal: true
    });
  };

  private handleHide = () => {
    this.setState({
      showModal: false
    });
  };

  private handleClose = () => {
    this.setState({
      showModal: false
    });
  };

  private toggleDomainRights = () => {
    this.setState(prevState => ({
      domainRightsAggrement: !prevState.domainRightsAggrement
    }));
  };

  private toggleAdditionalFee = () => {
    this.setState(prevState => ({
      additionalFeeAggrement: !prevState.additionalFeeAggrement
    }));
  };

  private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
    this.props.clearErrorMessage('externalDomain');
    this.setState({
      externalDomain: value
    });
  };

  private handleUpdateExternalDomain = () => {
    const instance = this.props.activeInstance;
    if (instance.data) {
      this.props.updateFieldValue(
        instance.data.id,
        'externalDomain',
        this.state.externalDomain
      );
    }
  };

  public render() {
    const instance = this.props.activeInstance;

    // Do not render if instance data is not present or if externalDomain is present
    // in instance data.
    if (!instance.data || (instance.data && instance.data.externalDomain)) {
      return null;
    }

    return (
      <div>
        <Button className="addBtn" size="lg" onClick={this.handleShow}>
          <WrappedMessage messages={messages} id="buttonText" />
        </Button>

        <Modal
          size="lg"
          show={this.state.showModal}
          onClose={this.handleClose}
          onHide={this.handleClose}
          centered
        >
          <Modal.Header className="add-domain-modal-header" closeButton />
          <Modal.Body>
            <Container className="add-domain-modal">
              <h2>
                <WrappedMessage messages={messages} id="modalTitle" />
              </h2>
              <Row>
                <Col className="add-domain-modal-description">
                  <p>
                    <span>
                      <WrappedMessage
                        messages={messages}
                        id="modalDescriptionOne"
                      />
                    </span>
                    <a
                      href="https://www.gandi.net/"
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      <WrappedMessage
                        messages={messages}
                        id="gandiDomainText"
                      />
                    </a>
                    <span>
                      <WrappedMessage
                        messages={messages}
                        id="modalDescriptionTwo"
                      />
                    </span>
                  </p>
                </Col>
              </Row>
              <Row>
                <Col lg={8}>
                  <TextInputField
                    value={this.state.externalDomain}
                    onChange={this.handleChange}
                    type="domain"
                    messages={messages}
                    fieldName="externalDomain"
                    autoComplete="off"
                  />
                  {instance.feedback.externalDomain ? (
                    <div className="domain-error">
                      <p className="error-msg">
                        <i className="fas fa-exclamation-triangle pr-2" />
                        <WrappedMessage
                          messages={messages}
                          id="domainErrorMessage"
                        />
                      </p>
                      <p className="error-detail">
                        <WrappedMessage
                          messages={messages}
                          id="domainErrorDescription1"
                        />
                        <a href={SUPPORT_LINK}>
                          <WrappedMessage
                            messages={messages}
                            id="contactSupport"
                          />
                        </a>
                        <WrappedMessage
                          messages={messages}
                          id="domainErrorDescription2"
                        />
                      </p>
                    </div>
                  ) : null}
                </Col>
              </Row>
              <Form.Group>
                <Form.Check type="checkbox" className="d-flex my-3">
                  <Form.Check.Input
                    type="checkbox"
                    checked={this.state.domainRightsAggrement}
                    onChange={this.toggleDomainRights}
                  />
                  <Form.Check.Label className="checkbox-label">
                    <WrappedMessage
                      messages={messages}
                      id="domainRightsAgreement"
                    />
                  </Form.Check.Label>
                </Form.Check>
                <Form.Check type="checkbox" className="d-flex my-3">
                  <Form.Check.Input
                    type="checkbox"
                    checked={this.state.additionalFeeAggrement}
                    onChange={this.toggleAdditionalFee}
                  />
                  <Form.Check.Label className="checkbox-label">
                    <WrappedMessage
                      messages={messages}
                      id="additionalFeeAgreement"
                    />
                  </Form.Check.Label>
                </Form.Check>
              </Form.Group>
              <div className="d-flex flex-row">
                <div className="verify-btn">
                  <Button
                    size="lg"
                    variant="primary"
                    disabled={
                      !(
                        this.state.additionalFeeAggrement &&
                        this.state.domainRightsAggrement &&
                        !!this.state.externalDomain
                      )
                    }
                    onClick={this.handleUpdateExternalDomain}
                  >
                    <WrappedMessage messages={messages} id="addDomainBtn" />
                  </Button>
                </div>
                <div>
                  <Button
                    size="lg"
                    variant="outline-primary"
                    onClick={this.handleHide}
                  >
                    Cancel
                  </Button>
                </div>
              </div>
            </Container>
          </Modal.Body>
        </Modal>
      </div>
    );
  }
}

export const AddDomainButton = connect<
  StateProps,
  ActionProps,
  {},
  Props,
  RootState
>((state: RootState) => state.console, {
  updateFieldValue,
  clearErrorMessage
})(AddDomainButtonComponent);
