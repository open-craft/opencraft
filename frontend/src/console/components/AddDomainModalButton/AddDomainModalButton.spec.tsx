import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { AddDomainButton } from './AddDomainModalButton';

it('renders without crashing', () => {
  const tree = setupComponentForTesting(
    <AddDomainButton />,
    {
      console: {
        loading: false,
        activeInstance: {
          data: {
            id: 1,
            instanceName: "test",
            subdomain: "test",
            draftThemeConfig: {
              version: 1,
              mainColor: "#444444",
              linkColor: "#FFAAFF"
            },
            draftStaticContentOverrides: {
              homepageOverlayHtml: "Test overlay",
            }
          },
          feedback: [],
          loading: ['draftThemeConfig'],
          deployment: null,
        },
      }
    }
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders null when externalDomain in data', () => {
  const tree = setupComponentForTesting(
    <AddDomainButton />,
    {
      console: {
        loading: false,
        activeInstance: {
          data: {
            id: 1,
            instanceName: "test",
            externalDomain: 'example.com',
            subdomain: "test",
            draftThemeConfig: {
              version: 1,
              mainColor: "#444444",
              linkColor: "#FFAAFF"
            },
            draftStaticContentOverrides: {
              homepageOverlayHtml: "Test overlay",
            }
          },
          feedback: [],
          loading: ['draftThemeConfig'],
          deployment: null,
        },
      }
    }
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
