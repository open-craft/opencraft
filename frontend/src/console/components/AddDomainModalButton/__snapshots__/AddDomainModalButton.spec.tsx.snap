// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders null when externalDomain in data 1`] = `null`;

exports[`renders without crashing 1`] = `
<div>
  <button
    className="addBtn btn btn-primary btn-lg"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add custom domain (â‚¬25/month)
  </button>
</div>
`;
